text/0003-execution-environment.md

```diff
@@ -0,0 +1,104 @@
+# Execution Environments
+
+## Summary
+
+This RFC introduces settings for controlling what execution environment (Node.js, Bun, Deno) will be used for a package during:
+
+* runnings its lifecycle scripts
+* building
+* running it as a CLI app
+
+## Motivation
+
+Running multiple versions of Node.js on the same computer isn't easy. Also, there is currently no way for a package to tell the package manager that it needs to be executed with a specific version of Node.js. Node.js versions should be locked the same way as other dependencies of projects are locked for reproducibility.
```

Comment:
There is already a way: engines

https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines

Perhaps this RFC could explain why engines is not sufficient and why we need a new feature.

Reply:
As you can see in the alternatives section:

> Instead of introducing a new field, we could use the `engines` field for detecting what Node.js version should be used for running the bin file or building the package. However, the `engines` field is already used by other package managers and it is usually just sets a range with the lowest supported Node.js version. If we will use it for specifying exact versions, installations of the package with other package managers will fail, when `engine-strict` is set to `true`.